catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -421,6 +450,156 @@ public void init() {
+
+            List<Metacard> fullList = new ArrayList<>();
+            fullList.addAll(results.stream()
+                    .map(Result::getMetacard)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()));
+
+            for (Metacard metacard : fullList) {
+                String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                        .getValue()
+                        .toString();
+                if (remoteRegistryMetacards.containsKey(registryId)) {
+                    Metacard currentMetacard = remoteRegistryMetacards.get(registryId);
+                    if (currentMetacard.getModifiedDate()
+                            .before(metacard.getModifiedDate())) {
+                        remoteRegistryMetacards.put(registryId, metacard);
+                    }
+                } else {
+                    remoteRegistryMetacards.put(registryId, metacard);
+                }
+            }
+
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+            String message = "Error querying for subscribed metacards;";
+            LOGGER.error("{} For source ids: {}", message, pollableSourceIds);
+            throw new FederationAdminException(message, e);
+        }
+
+        return remoteRegistryMetacards;
+    }
+
+    private Map<String, Metacard> getNonLocalRegistryMetacardsMap()
+            throws FederationAdminException {
+
+        Map<String, Metacard> nonLocalMetacardsMap = new HashMap<>();
+        List<Filter> filters = new ArrayList<>();
+        filters.add(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.CONTENT_TYPE),
+                RegistryConstants.REGISTRY_NODE_METACARD_TYPE_NAME));
+        filters.add(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.TAGS),
+                RegistryConstants.REGISTRY_TAG));
+        filters.add(filterBuilder.not(filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE)
+                .is()
+                .bool(true)));
+
+        Filter filter = FILTER_FACTORY.and(filters);
+
+        List<Metacard> nonLocalMetacardList = getRegistryMetacardsByFilter(filter);
+
+        for (Metacard metacard : nonLocalMetacardList) {
+            String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .getValue()
+                    .toString();
+            nonLocalMetacardsMap.put(registryId, metacard);
+        }
+
+        return nonLocalMetacardsMap;
+    }
+
+    private Map<Serializable, Metacard> getRegistryUpdatesForRemoteRegistries()
+            throws FederationAdminException {
+        Map<String, Metacard> remoteRegistryMetacardsMap = getRemoteRegistryMetacardsMap();
+        Map<String, Metacard> nonLocalMetacardsMap = getNonLocalRegistryMetacardsMap();
+
+        Map<Serializable, Metacard> updateMap = new HashMap<>();
+
+        remoteRegistryMetacardsMap.entrySet()
```

Comment:
This looks like it handles updating existing metacards but what about new metacards? we need to create them in our local catalog if we don't have them.